npm/pkg/dataplane/dataplane_test.go

```diff
@@ -9,14 +9,145 @@ import (
 
 func TestNewDataPlane(t *testing.T) {
 	metrics.InitializeAll()
-	dp := NewDataPlane()
+	dp := NewDataPlane("testnode")
 
 	if dp == nil {
 		t.Error("NewDataPlane() returned nil")
 	}
 
-	err := dp.CreateIPSet("test", ipsets.NameSpace)
+	dp.CreateIPSet("test", ipsets.NameSpace)
+}
+
+func TestInitializeDataPlane(t *testing.T) {
```

Comment:
minor code style note: we can write the following code in a TestMain() function (called before all testing begins) and then reset the dataplane at the top of each test to prevent having this in every test:
```
	metrics.InitializeAll()
	dp := NewDataPlane("testnode")

	if dp == nil {
		t.Error("NewDataPlane() returned nil")
	}

	err := dp.InitializeDataPlane()
	if err != nil {
		t.Errorf("InitializeDataPlane() returned error %v", err)
	}
```